### DISCUSSION TITLE
Debian OS packages truly needed

### DISCUSSION THREAD

**Discussion by alfredo-rico** [libvips/ruby-vips#405] [Q&A] [answered]
Hi :wave: 
For a Debian/Ubuntu OS I've read a post saying that the packages needed for ruby-vips (to be used in rails project) are:
```
sudo apt install libvips42 libvips-dev libvips-tools 
```
But do we really need all of them? I'd like to confirm if just installing `libvips42` is good enough. My rails project  doesn't need shell scripting and `ruby-vips` has been already installed as a dependency. So no gem compilation needed I think. 

Thanks

**Comment by jcupitt** [comment, marked as answer]
Hello @alfredo-rico,

Yes, just:

```
sudo apt install libvips42 --no-install-recommends
```

The `--no-install-recommends` makes it usefully smaller again:

```
$ docker run --rm -it ubuntu:24.04
root@62a2ae7f9b8c:/# apt update && apt upgrade
...
root@62a2ae7f9b8c:/# apt install libvips42 --no-install-recommends
Reading package lists... Done
...
After this operation, 132 MB of additional disk space will be used.
...
```

You can make it smaller still if you build your own libvips binary.

I would also look at the checklist here:

https://github.com/libvips/libvips/blob/master/doc/Developer-checklist.md

Particularly:

https://github.com/libvips/libvips/blob/master/doc/Developer-checklist.md#only-enable-the-load-libraries-you-need
